sdk/tables/Azure.Data.Tables/src/Queryable/ExpressionWriter.cs

```diff
@@ -199,9 +201,18 @@ private string Translate(Expression e)
             return _builder.ToString();
         }
 
-        protected virtual string TranslateMemberName(string memberName)
+        protected virtual string TranslateMemberName(MemberInfo memberName)
         {
-            return memberName;
+            PropertyInfo partitionKeyProperty = memberName.DeclaringType
+                .GetProperties()
+                .FirstOrDefault(p => p.Name == memberName.Name && Attribute.IsDefined(p, typeof(DataMemberAttribute)));
```

Comment:
It would be great if we could cache this reflection since we do it for each property in the expression unconditionally.